pkg/labeler.go

```diff
@@ -68,6 +74,54 @@ func NewBranchCondition() Condition {
 	}
 }
 
+func NewFilesCondition(pr *gh.PullRequest) Condition {
+	ghToken := os.Getenv("GITHUB_TOKEN")
+	diffReq, _ := http.NewRequest("GET", pr.GetDiffURL(), nil)
+	diffReq.Header.Add("Authorization", "Bearer "+ghToken)
+	diffRes, _ := http.DefaultClient.Do(diffReq)
+
+	defer diffRes.Body.Close()
+
+	var diffRaw []byte
+	prFiles := make([]string, 0)
+	if diffRes.StatusCode == http.StatusOK {
+		diffRaw, _ = ioutil.ReadAll(diffRes.Body)
+
+		diff, _ := diffparser.Parse(string(diffRaw))
+		prFilesSet := map[string]struct{}{}
+		for _, file := range diff.Files {
+			prFilesSet[file.OrigName] = struct{}{}
+			prFilesSet[file.NewName] = struct{}{}
+		}
+		for k := range prFilesSet {
+			prFiles = append(prFiles, k)
+		}
+	}
```

Comment:
Could we move this code into `Evaluate` in the `Condition`?  I created the `Condition` objects to encapsulate the computation but avoid running any of it unit they are processed by `findMatches`.  Part of the reason is to do all error handling there.

Reply:
👍   One side effect of this will be it will download and parse the patch for every label that uses a file match. Potentially not that big of a deal but if there were a lot of files changed in a PR this could be wasteful